### PR TITLE
Polygon

### PR DESCRIPTION
**Pull request by Rotwitt**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

Adding Polygon 65 to the QMK master

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/beak65/rules.mk [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+# Build Options
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = no       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (waffle87) [inline keyboards/beak65/rules.mk:15]:
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable bootmagic lite
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
```

File: keyboards/beak65/via.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
+{
```
Comment (waffle87) [inline keyboards/beak65/via.json:1]: This file needs to be removed.

File: keyboards/polygon/ansi/config.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+#include "config_common.h"
+
+#define DRIVER_ADDR_1 0b0110000
+#define DRIVER_ADDR_2 0b0110000 // this is here for compliancy reasons.
+
+#define DRIVER_COUNT 1
+#define DRIVER_1_LED_TOTAL 66
+#define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
+
+#define RGB_DISABLE_TIMEOUT 20000
```
Comment (waffle87) [inline keyboards/polygon/ansi/config.h:29]:
```suggestion
#define RGB_MATRIX_LED_COUNT DRIVER_1_LED_TOTAL

#define RGB_MATRIX_TIMEOUT 20000
```

File: keyboards/polygon/ansi/rules.mk [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+# Build Options
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (waffle87) [inline keyboards/polygon/ansi/rules.mk:15]: This file should be left empty unless it overrides anything in `keyboards/polygon/rules.mk`
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable bootmagic lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
```

File: keyboards/polygon/config.h [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x6959 //
+#define PRODUCT_ID   0x065F //
+#define DEVICE_VER   0x0001 // rev1
+#define MANUFACTURER rotwitt
+#define PRODUCT      polygon
```
Comment (waffle87) [inline keyboards/polygon/config.h:27]: This needs to be moved to `info.json`.

File: keyboards/polygon/config.h [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+/* key matrix size */
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 16
+
```
Comment (waffle87) [inline keyboards/polygon/config.h:32]:
```suggestion
```

File: keyboards/polygon/config.h [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
+#define NO_ACTION_FUNCTION
+
+/* Bootmagic Lite key configuration */
+//#define BOOTMAGIC_LITE_ROW 0
+//#define BOOTMAGIC_LITE_COLUMN 0
```
Comment (waffle87) [inline keyboards/polygon/config.h:115]:
```suggestion
```

File: keyboards/polygon/info.json [hunk 8]
```diff
@@ -0,0 +1,84 @@
+{
+    "keyboard_name": "polygon",
+    "url": "https://www.instagram.com/byrotwitt/",
+    "maintainer": "trip-trap",
+    "width": 15,
+    "height": 5,
+    "layouts": {
+      "LAYOUT_ansi": {
```
Comment (waffle87) [inline keyboards/polygon/info.json:8]:
```suggestion
    "processor": "atmega32u4",
    "bootloader": "atmel-dfu",
    "layouts": {
      "LAYOUT": {
```

File: keyboards/polygon/iso/config.h [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+#include "config_common.h"
+
+#define DRIVER_ADDR_1 0b0110000
+#define DRIVER_ADDR_2 0b0110000 // this is here for compliancy reasons.
+
+#define DRIVER_COUNT 1
+#define DRIVER_1_LED_TOTAL 67
+#define DRIVER_LED_TOTAL DRIVER_1_LED_TOTAL
+
+#define RGB_DISABLE_TIMEOUT 20000
```
Comment (waffle87) [inline keyboards/polygon/iso/config.h:29]:
```suggestion
#define DRIVER_COUNT 1
#define DRIVER_1_LED_TOTAL 67
#define RGB_MATRIX_LED_COUNT DRIVER_1_LED_TOTAL

#define RGB_MATRIX_TIMEOUT 20000
```

File: keyboards/polygon/iso/rules.mk [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+# Build Options
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (waffle87) [inline keyboards/polygon/iso/rules.mk:15]: This file should be left empty unless it overrides anything in `keyboards/polygon/rules.mk`
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable bootmagic lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
```

File: keyboards/beak65/beak65.h [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    K400, K401, K402,                   K406,                   K410, K411, K412, K413, K414, K415, K416 \
+) { \
+    { K000, K001, K002, K003, K004, K005, K006, K007, K008, K009, K010, K011, K012, K013, ____, ____, ____ }, \
+    { K100, K101, K102, K103, K104, K105, K106, K107, K108, K109, K110, K111, K112, K113, ____, KENC, ____ }, \
+    { K200, K201, K202, K203, K204, K205, K206, K207, K208, K209, K210, K211, ____, K213, ____, ____, ____ }, \
+    { K300, ____, K302, K303, K304, K305, K306, K307, K308, K309, K310, K311, ____, K313, ____, K315, ____ }, \
+    { K400, K401, K402, ____, ____, ____, K406, ____, ____, ____, K410, K411, K412, K413, K414, K415, K416 }  \
+}
+
+#define LAYOUT_via( \
```
Comment (waffle87) [inline keyboards/beak65/beak65.h:37]:
```suggestion
#define LAYOT_all( \
```

File: keyboards/polygon/keymaps/ansi/keymap.c [hunk 12]
```diff
@@ -0,0 +1,54 @@
+/* Copyright 2020 Adam Naldal
```
Comment (waffle87) [inline keyboards/polygon/keymaps/ansi/keymap.c:1]: You need a `default` keymap.

File: keyboards/polygon/keymaps/via/config.h [hunk 13]
```diff
@@ -0,0 +1,25 @@
+/* Copyright 2020 Steven Nguyen and Adam Naldal
```
Comment (waffle87) [inline keyboards/polygon/keymaps/via/config.h:1]: This file should be removed.

File: keyboards/polygon/keymaps/via/keymap.c [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    [1] = LAYOUT_via(
+     KC_GRV,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10,  KC_F11,  KC_F12, _______,
+  	_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,           _______,
+  	_______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+  	_______,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,                 _______,             MO(2),
+  	_______, _______, _______,                            _______,                            _______, _______, _______, _______,  _______, _______, _______,
+//  encoder codes:
+    KC_WH_U, KC_WH_D),
+    [2] = LAYOUT_via(
+      RESET, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
```
Comment (waffle87) [inline keyboards/polygon/keymaps/via/keymap.c:38]:
```suggestion
      QK_BOOT, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
```

File: keyboards/polygon/keymaps/via/keymap.c [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+void encoder_update_user(uint8_t index, bool clockwise) {
+    uint8_t layer = get_highest_layer(layer_state);
+    if (index == 0) {
+        if (clockwise) {
+            tap_code16(dynamic_keymap_get_keycode(layer, 5, 1));
+        } else {
+            tap_code16(dynamic_keymap_get_keycode(layer, 5, 0));
+        }
+    }
+}
```
Comment (waffle87) [inline keyboards/polygon/keymaps/via/keymap.c:64]: Encoder map should be used here.
```suggestion
```

File: keyboards/polygon/keymaps/via/rules.mk [hunk 16]
```diff
@@ -0,0 +1,10 @@
+LEADER_ENABLE = no
+FAUXCLICKY_ENABLE = no
+MIDI_ENABLE = no
+BLUETOOTH_ENABLE = no
+KEY_LOCK_ENABLE = no
+TERMINAL_ENABLE = no
+LTO_ENABLE = yes
+LEADER_ENABLE = no
+
```
Comment (waffle87) [inline keyboards/polygon/keymaps/via/rules.mk:9]:
```suggestion
LTO_ENABLE = yes
```

File: keyboards/polygon/polygon.h [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define ____ KC_NO
+
+#define LAYOUT_ansi( \
```
Comment (waffle87) [inline keyboards/polygon/polygon.h:23]: See above comment about layout naming conventions.

File: keyboards/polygon/readme.md [hunk 18]
```diff
@@ -0,0 +1,29 @@
+# Polygon
+
+![Polygon]()
```
Comment (waffle87) [inline keyboards/polygon/readme.md:3]:
```suggestion
```

File: keyboards/polygon/rules.mk [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+
+# Build Options
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (waffle87) [inline keyboards/polygon/rules.mk:15]:
```suggestion
BOOTMAGIC_ENABLE = yes     # Enable bootmagic lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
```

File: keyboards/polygon/via.json [hunk 20]
```diff
@@ -0,0 +1,148 @@
+{
```
Comment (waffle87) [inline keyboards/polygon/via.json:1]: This file should be removed.

**Review by waffle87 — CHANGES_REQUESTED** [review]
You clearly did not read [PR checklist](https://docs.qmk.fm/#/pr_checklist) or even test your changes before opening this PR. Please show a little respect for the project and do so in the future.

**Review by Rotwitt — COMMENTED** [review]
Sorry! 
I had the files made for me and presumed they followed the guidelines.
I'll review your changes, and be more careful in the feature.

**Pull request closed without merging**